src/voting/CIP-voting.md

```diff
@@ -0,0 +1,166 @@
+
+These definitions extend [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/) to provide specific support for vote delegation.
+
+# **Motivation**
+The goal for this CIP is to extend the dApp-Wallet web bridge to enable the construction of transactions containing metadata that conforms to
+[CIP-36 (Catalyst/Voltaire Registration Transaction Metadata Format - Updated)](https://cips.cardano.org/cips/cip36/),
+enabling new functionality including vote delegation to either private or public representatives (dReps),
+splitting or combining of private votes,
+the use of different voting keys or delegations
+for different purposes (Catalyst etc).
+
+# **Specification**
+
+## `Types`
+
+### **GovernanceKey**
+
+```
+type GovernanceKey = {
+  votingKey: string,
+  weight: number
+}
+
+```
+
+`votingKey`: Ed25519 pubkey 32 bytes HEX string  
+
+`weight`: Used to calculate the actual voting power using the rules described
+in 
+[CIP-36](https://cips.cardano.org/cips/cip36/).
+
+
+### **Purpose**
+
+```
+type enum Purpose = {
+  CATALYST = 0,
+  OTHER = 1
+}
+
+```
+
+`Purpose`: Defines the purpose of the delegations. This is used to limit the scope of the delegations.  For example, a purpose might be a subset of Catalyst proposals, a council election, or even some private purpose (agreed by convention).
+
+## **Namespace**
+
+### **cardano.{walletName}.governance.enable(): Promise\<API>**
+The `cardano.{walletName}.governance.enable()` method is used to enable the governance API. It should request permission from the wallet to enable the API. If permission is granted, the rest of the API will be available. The walelt may choose to maintain a whitelist of allowed clients to avoid asking for permission every time.
+
+This api being an extension of [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/), expects that `cardano.{walletName}.enable()` to be enabled implicitly. 
+
+# **`API`**
+
+## **api.getVotingKey**(address_index: number = 0, account: number = 0, role: number = 0): Promise\<Bip32PublicKey>
+
+Should derive and return the wallets voting public key
+
+### **Returns**
+`cbor\<vkey>` - cbor serialized 64 bytes (x, y) Ed25519 public key  
+
+There should be only a single single voting key per wallet / mnemonic. 
+
+The **`votingKey`** should be derived from the following path. 
+
+```
+m / 1694' / 1815' / account' / role' / address_index'
+```
+
+`1694` (year Voltaire was born) Sets a dedicated `purpose` in the derivation path for the voting profile.  
+
+`address_index` - index of the key to use. 
+
+
+## **api.buildDelegation**(keys: GovernanceKey[], purpose: Purpose, networkId: number, stakeAccountPath: number = 0, stakeRolePath: number = 0, stakeIndex: number = 0): Promise\<**`Delegation`**>
```

Comment:
A few questions here:

> networkId: number

While it's needed to derive reward address, wallet is usually connected to a single network at a time, so this argument would probably be used mostly for validation. Maybe document that an APIError can be thrown here if wallet is currently running under a different networkId? Also, consider using [CIP-34](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0034) to identify the network more accurately ("not just any testnet")

> stakeAccountPath: number = 0, stakeRolePath: number = 0, stakeIndex: number = 0

Why would a dApp want to specify a specific stake key derivation path? Shouldn't it be up the wallet to decide which stake key to use for the given purpose? Wallet might want to present some UI for user to choose if there are multiple.